ui/src/app/pages/login/account-creation/account-creation.tsx

```diff
@@ -0,0 +1,665 @@
+      },
+      institutionName, institutionRole, showNext, affiliation, affiliationRole, affiliationOther
+    } = this.state;
+    return <div id='account-creation'
+                style={{'paddingTop': environment.enableAccountPages ? '1.5rem' :
+                      '3rem', 'paddingRight': '3rem', 'paddingLeft': '3rem'}}>
+      <Header>Create your account</Header>
+      {environment.enableAccountPages && !showNext && <div style={{marginTop: '0.5rem'}}>
+        <label style={{color: colors.primary, fontSize: 16}}>
+          Please complete Step 1 of 2
+        </label>
+        {this.state.errors && <div className='error-messages'>
+          <ValidationError>
+            {summarizeErrors(this.state.errors)}
+          </ValidationError>
+        </div>}
+        <Section header='Create an All of Us username'>
+          <div>
+            <TextInput id='username' name='username' placeholder='New Username'
+                       value={username}
+                       onChange={v => this.usernameChanged(v)}
+                       invalid={this.state.usernameConflictError || this.usernameInvalidError()}
+                       style={{width: '12rem', marginRight: '0.5rem'}}/>
+            <div style={inputStyles.iconArea}>
+              <ValidationIcon validSuccess={this.usernameValid()}/>
+            </div>
+            <TooltipTrigger content={<div>Usernames can contain only letters (a-z),
+              numbers (0-9), dashes (-), underscores (_), apostrophes ('), and periods (.)
+              (minimum of 3 characters and maximum of 64 characters).<br/>Usernames cannot
+              begin or end with a period (.) and may not contain more than one period (.) in a row.
+            </div>}>
+              <InfoIcon style={{'height': '22px', 'paddingLeft': '2px'}}/>
+            </TooltipTrigger>
+            {this.state.usernameConflictError &&
+            <div style={{height: '1.5rem'}}>
+              <Error id='usernameConflictError'>
+                Username is already taken.
+              </Error></div>}
+            {this.usernameInvalidError() &&
+            <div style={{height: '1.5rem'}}><Error id='usernameError'>
+              Username is not a valid username.
+            </Error></div>}
+          </div>
+        </Section>
+        <Section header='About you'>
+          <div style={{display: 'flex', flexDirection: 'column'}}>
+            <div style={{paddingBottom: '0.5rem'}}>
+              <TextInput id='givenName' name='givenName' autoFocus
+                         placeholder='First Name'
+                         value={givenName}
+                         invalid={givenName.length > 80}
+                         style={{width: '12rem', marginRight: '2rem'}}
+                         onChange={v => this.updateProfile('givenName', v)}/>
+              {givenName.length > 80 &&
+              <ErrorMessage id='givenNameError'>
+                First Name must be 80 characters or less.
```

Comment:
If we parameterize the name length limit, these error messages can also read text from that.

Reply:
I think you could change 80 to `{{nameLength}}` (and below as well)